docs/src/pages/components/data-grid/rendering/InfiniteLoadingGrid.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { DataGrid, GridOverlay } from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import LinearProgress from '@material-ui/core/LinearProgress';
+
+function CustomLoadingOverlay() {
+  return (
+    <GridOverlay>
+      <div style={{ position: 'absolute', top: 0, width: '100%' }}>
+        <LinearProgress />
+      </div>
+    </GridOverlay>
+  );
+}
+
+export default function InfiniteLoadingGrid() {
+  const [size, setSize] = React.useState(20);
```

Comment:
Maybe we should rename it?

```suggestion
  const [rowLength, setRowLength] = React.useState(20);
```

Reply:
I can do `rowLengthSize` because `setRowLength` clashes with the one have from the `useDemoData`.

Reply:
I've removed it all together as we can get that value from the props.